src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11057,67 +11102,28 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
                 }
                 else
                 {
-                    break;
+                    // Something that doesn't expand the current expression we're looking at.  Bail out and see if we
+                    // can end with a conditional expression.
+                    return false;
                 }
```

Comment:
note: we could consider extracing this into a tryGetOperatorKind helper, to get similar benefits on single-exit logic.  not sure if it is needed thoughl

Reply:
What I am finding as a reviewer is that the repeated "if-return-if-return-if-return"s are much easier to understand at a glance than the repeated "if-else-if-else-if-else-if". Both here and below, where the `leftOperand` is assigned.

This is in part because the mental compiler needs to locate where we are going to jump to after exiting a large if-else block, scroll to it, decide whether any additional work is being done, decide whether we are in a consistent state for doing that work, etc.

Also, I don't think there's anyway to "fix" this, but, I am finding that scrolling between corresponding places in the old and new versions, to find where logic has moved to, is quite challenging, to the point that I might suggest to the second reviewer to just skim thru the deleted lines, then review the new change practically as new code.